File: python/raindrops/raindrops.py
# convert converts the sound of 'number' of raindrops. The function return the
# sound in a string format
def convert(number):
    sound = ''
    if number % 3 != 0 and number % 5 != 0 and number % 7 != 0:
        sound = str(number)
    if number % 3 == 0:
        sound += "Pling"
    if number % 5 == 0:
        sound += "Plang"
    if number % 7 == 0:
        sound += "Plong"

    return sound

File: go/raindrops/raindrops.go
// Package raindrops provides functionality to convert a number to a raindrop
// sound
package raindrops

import "strconv"

// Convert converts a number to its sound
func Convert(number int) string {

	var raindropSound string

	var numberToSoundConverter = []struct {
		soundNumber int
		sound       string
	}{
		{3, "Pling"},
		{5, "Plang"},
		{7, "Plong"},
	}

	for _, conversionElement := range numberToSoundConverter {
		if number%conversionElement.soundNumber == 0 {
			raindropSound += conversionElement.sound
		}
	}

	if raindropSound != "" {
		return raindropSound
	}

	return strconv.Itoa(number)
}
